### PR TITLE
fix(bug): fix rare hail overwrite

### PR DESCRIPTION
**Pull request by flowers9**
**Bugfix:** This PR addresses issue #9060

## Fix Details
Based on Amazinite's comment, I found one place in the human hails that could randomly put a space at the end of a hail.  I moved it so that wouldn't happen.

I also think I found the code that fails to advance the line when it deals with a text string that ends in a non-CR whitespace.  It now checks to see if the line since the last advance has any non-whitespace, and advances the line even if it's not in a word at the moment.

## Testing Done
Modified human hails so most navy hails ended in a space. Hung in Sol on fast-forward and watched hails.

Edit: found the code that generates the issue, not just a hail that triggers it.

### CODE REVIEW

**Review by OcelotWalrus — APPROVED** [review]
I tried the code and it seems to fix the issue, nice work here!

File: source/text/WrappedText.cpp [hunk 1]
```diff
@@ -294,6 +298,10 @@ void WrappedText::Wrap()
 		word.x += width;
 		lineWidth = word.x;
 	}
+	// If we are not in a word, advance to the next line if this line has any words.
```
Comment (Amazinite) [inline source/text/WrappedText.cpp:301]: Took me a few minutes of reading through this code to figure out how this fixed the bug. I think a slight change to this comment would help make it more clear what's happening.
```suggestion
	// If the text ended in whitespace but the line did contain a word at some point,
	// then we must still advance to the next line.
```
Maybe also update the comment `// Add this final word to the existing words.` a few lines above this to read `// Add this final word to the existing words and advance to the next line.`, using the "advance to the next line" vocab in both places to make a connection between what normally happens and this special case.

**Comment by Amazinite** [comment]
Nice find. :+1:

File: source/text/WrappedText.cpp (hunk 1 above)
Comment (flowers9) [reply source/text/WrappedText.cpp]: You're not wrong - this section of code could use a bit more clarity.  I still get thrown by AdjustLine being the thing that sets lineBegin and lineWidth.

**Pull request merged**